FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SNTPClientTask.c

```diff
@@ -0,0 +1,793 @@
+
+        /* Check the socket was created successfully. */
+        /* TODO - Consider using random port assigned by FreeRTOS for better protection */
+        /* against "off-path" attacker. */
+        if( pUdpContext->socket == FREERTOS_INVALID_SOCKET )
+        {
+            /* There was insufficient FreeRTOS heap memory available for the socket
+             * to be created. */
+            LogError( ( "Failed to create UDP socket for SNTP client due to insufficient memory." ) );
+        }
+        else
+        {
+            struct freertos_sockaddr bindAddress;
+            UdpTransportInterface_t udpTransportIntf;
+
+            bindAddress.sin_port = FreeRTOS_htons( SNTP_DEFAULT_SERVER_PORT );
+
+            if( FreeRTOS_bind( pUdpContext->socket, &bindAddress, sizeof( bindAddress ) ) == 0 )
+            {
+                /* The bind was successful. */
+                LogDebug( ( "UDP socket has been bound to port %lu", SNTP_DEFAULT_SERVER_PORT ) );
+            }
+
+            /* Set the UDP transport interface object. */
+            udpTransportIntf.pUserContext = pUdpContext;
+            udpTransportIntf.sendTo = UdpTransport_Send;
+            udpTransportIntf.recvFrom = UdpTransport_Recv;
+
+            /* Initialize context. */
+            Sntp_Init( pContext,
+                       pServers,
+                       numOfServers,
+                       democonfigSERVER_RESPONSE_TIMEOUT_MS,
+                       pContextBuffer,
+                       contextBufferSize,
+                       resolveDns,
+                       sntpClient_GetTime,
+                       sntpClient_SetTime,
+                       &udpTransportIntf,
+                       NULL );
+
+            initStatus = true;
+        }
+    }
+
+    return initStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+void sntpTask( void * pParameters )
+{
+    SntpContext_t clientContext;
+    bool initStatus = false;
+
+    /* Variable representing the SNTP client context. */
+    static SntpContext_t context;
+
+    /* Memory for the SNTP packet buffer in the SNTP context. */
+    static uint8_t contextBuffer[ SNTP_CONTEXT_NETWORK_BUFFER_SIZE ];
+
+    /* Memory for the network context representing the UDP socket that will be
+     * passed to the SNTP client context. */
+    static NetworkContext_t udpContext;
+
+    /* Store the configured time servers in an array. */
+    static const char * pTimeServers[] = { democonfigLIST_OF_TIME_SERVERS };
+    const size_t numOfServers = sizeof( pTimeServers ) / sizeof( char * );
+
+    initStatus = initializeSntpClient( &clientContext,
+                                       pTimeServers,
+                                       numOfServers,
+                                       contextBuffer,
+                                       sizeof( contextBuffer ),
+                                       &udpContext );
+
+    if( initStatus == true )
```

Comment:
If `initStatus` is false, this task will exit which is an error.

Reply:
Added a `configASSERT` to enable the developer to detect the failed initialization status as we discussed offline.